opencti-platform/opencti-graphql/src/database/middleware.js

```diff
@@ -2375,8 +2398,22 @@ const upsertElement = async (context, user, element, type, basePatch, opts = {})
     const fileImpact = { key: 'x_opencti_files', value: [...(element.x_opencti_files ?? []), convertedFile] };
     inputs.push(fileImpact);
   }
-  // If confidence is passed at creation, just compare confidence
-  const isConfidenceMatch = (updatePatch.confidence ?? 0) >= (element.confidence ?? 0);
+
+  // region confidence control
+  if (!user.effective_confidence_level) {
+    // LockTimeoutError would make the worker retry (we do not want to lose messages in that case
+    // TODO: new error and update worker?
+    throw LockTimeoutError({ user }, 'User has no effective confidence level and cannot upsert data in the platform.');
+  }
+  const elementConfidence = Math.max(Math.min(element.confidence ?? 0, 100), 0);
```

Comment:
Looks a bit difficult to read.

Reply:
Once https://github.com/OpenCTI-Platform/opencti/pull/5787 is merged, I'll replace this line with a more readable `cropNumber(value, 0, 100)`